src/sql/src/pure.rs

```diff
@@ -162,9 +162,29 @@ pub async fn purify(mut stmt: Statement<Raw>) -> Result<Statement<Raw>, anyhow::
             Connector::PubNub { .. } => (),
         }
 
-        purify_format(format, connector, col_names, file, &config_options).await?;
-        if let sql_parser::ast::Envelope::Upsert(format) = envelope {
-            purify_format(format, connector, col_names, None, &config_options).await?;
+        purify_format(
+            format,
+            connector,
+            &envelope,
+            col_names,
+            file,
+            &config_options,
+        )
+        .await?;
+        if let sql_parser::ast::Envelope::Upsert(_) = envelope {
```

Comment:
the problem is that format needs to be `&mut` -- because it is mutated -- so both you need to pass the original format in, and you run into lifetime issues if you don't clone the envelope. Since the envelope is not mutated, it's safe to pass in a dupe of it. I'll add a comment here.